packages/server/modules/workspaces/repositories/workspaces.ts

```diff
@@ -102,3 +105,11 @@ export const upsertWorkspaceRoleFactory =
       .onConflict(['userId', 'workspaceId'])
       .merge(['role'])
   }
+
+export const getWorkspaceProjectsFactory =
```

Comment:
Not to be that guy, but this doesn't seam to have integration tests

Reply:
Also, i think it would be probably better to reuse the existing project repository function by extending it with an optional `workspaceId` argument. The benefit there would be, that the project list functionality will be consistent both in and outside of a workspace.

Reply:
no, fair. I like extending the existing repo functions, but I'm not thrilled by the two options (I found):

- `getDiscoverable` is too focused on _public_ projects
- `getUser` requires a user, which we can do here but may not always be the case

I gave it a swing, though. What do you think? If you're ok with this strategy I'll come back in and add the tests.